src/cmd-buildextend-azure

```diff
@@ -0,0 +1,193 @@
+                "path": os.path.basename(tmp_img_azure_vhd),
+                "size": int(fsize)
+        }
+        log.debug(
+            " * size is %s",
+            self._found_files[tmp_img_azure_vhd]["size"])
+        # ---
+
+
+def cli():
+    """
+    Parse args and dispatch
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build', help='Build ID', required=True)
+    parser.add_argument(
+        '--container', help='Storage location to write to',
+        required=True, default=os.environ.get("AZURE_CONTAINER"))
+    parser.add_argument(
+        '--force', help='Replace existing images and upload',
+        action='store_true',
+        default=bool(os.environ.get('AZURE_FORCE', False)))
+    parser.add_argument(
+        '--storage-account', help='Storage account', required=True,
+        default=os.environ.get('AZURE_STORAGE_ACCOUNT'))
+    parser.add_argument(
+        '--profile', help='Path to Azure profile',
+        required=True, default=os.environ.get('AZURE_PROFILE'))
+    args = parser.parse_args()
+
+    # Argument checks for environment strings that are required
+    arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
+    if args.profile is None:
+        raise Exception(arg_exp_str.format('profile', 'AZURE_PROFILE'))
+    if args.container is None:
+        raise Exception(arg_exp_str.format('container', 'AZURE_CONTAINER'))
+    if args.storage_account is None:
+        raise Exception(
+            arg_exp_str.format('storage_account', 'AZURE_STORAGE_ACCOUNT'))
+
+    # Identify the builds
+    build = Build(os.path.join('builds', args.build), args.build)
+
+    if 'azure' in build.meta['images'] and not args.force:
+        print('Azure image already built; use --force to rebuild')
+        raise SystemExit()
+
+    build.build_artifacts(cli_args=args)
+    run_ore(args, build)
+
+
+def run_ore(args, build):
+    """
+    Execute ore to upload the image and register the image
+    See:
+      - https://github.com/coreos/mantle/#azure
+      - https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction
+
+    TODO: Move to using an _Upload subclass.
+
+    :param args: The command line arguments
+    :type args: argparse.Namespace
+    :param build: Build instance to use
+    :type build: Build
+    """
+    # Stage 1: Upload blob
+    blob_name = build.azure_vhd_name
+    azure_vhd_path = os.path.join(TMPDIR, build.azure_vhd_name)
+    ore_upload_args = [
+        'ore', 'azure', 'upload-blob', args.storage_account,
+        args.container, blob_name, azure_vhd_path, '--validate',
+        '--azure-profile', args.profile]
+    if args.force:
+        ore_upload_args.append('--overwrite')
+    run_verbose(ore_upload_args)
+
+    # Stage 2: Turn the blob into an image
+    ore_create_image_args = [
+        'ore', 'azure', 'create-image',
+        '--azure-profile', args.profile,
+        '--blob', blob_name,
```

Comment:
I haven't personally tested the ASM commands but at least for the ARM based commands (which are part of https://github.com/coreos/mantle/pull/771 and likely what you will want to ultimately use) the entire URL is required to be input to the `--blob` parameter of `create-image`.  These URLs are predictable (`https://$storage_account.blob.core.windows.net/$container/$blob_name`) or can be parsed from the output of the `upload-blob*` commands. The `upload-blob-arm` command prints JSON (in the format `{"url": "$url"}`) and it looks like `upload-blob` prints `Blob uploaded to $url`, so maybe it's better to just build the URL manually.

Reply:
Updated